src/sonar_widget/SonarPlot.cc

```diff
@@ -206,6 +225,16 @@ void SonarPlot::sonarPaletteChanged(int index){
     applyColormap((ColorGradientType) index);
 }
 
+// update the sector scan (for scanning sonars)
+void SonarPlot::setSectorScan(bool continuous, base::Angle left, base::Angle right){
+    this->continuous = continuous;
+    this->leftLimit = left;
+    this->rightLimit = right;
+
+    if (!continuous)
+        changedSectorScan = true;
```

Comment:
don't you need to set this flag if you switch from non-continuous to continuous ? (in which case, set it to true in all cases, it won't happen that often anyways)